### PR TITLE
Fix the nan issue when enabled avx512f flags and run ci testing

### PR DESCRIPTION
**Pull request by yihuaxu**
When run ci testing, it will show the nan issue after enabled avx512f flags. Just updated the latest version of eigen, it can be fixed.

Relevant issue:  [ISSUE#14040](https://github.com/PaddlePaddle/Paddle/issues/14040)
Relevant PR:  [PR#14041](https://github.com/PaddlePaddle/Paddle/pull/14041)

### CODE REVIEW

**Comment by yihuaxu** [comment]
@tensor-tang Please help me review this commit. 
thanks a lot!

File: cmake/external/eigen.cmake [hunk 1]
```diff
@@ -32,7 +32,7 @@ else()
         GIT_REPOSITORY  "https://github.com/eigenteam/eigen-git-mirror"
         # eigen on cuda9.1 missing header of math_funtions.hpp
         # https://stackoverflow.com/questions/43113508/math-functions-hpp-not-found-when-using-cuda-with-eigen
-        GIT_TAG         917060c364181f33a735dc023818d5a54f60e54c
+        GIT_TAG         a826c4090d3b8af63c2a9b85e88a3b496eb511e1
```
Comment (luotao1) [inline cmake/external/eigen.cmake]: 请问为什么要更新TAG？

Comment (yihuaxu) [reply cmake/external/eigen.cmake]: @luotao1  因为在开启AVX512的时候，EIGEN会使能EIGEN_FAST_MATH。低版本会出现resnet50等测试项不能通过，更新到最新版本可以通过。

**Comment by luotao1** [comment]
请在commit信息里加`test=develop`来触发CI

File: cmake/external/eigen.cmake (hunk 1 above)
Comment (luotao1) [reply cmake/external/eigen.cmake]: > 因为在开启AVX512的时候，EIGEN会使能EIGEN_FAST_MATH。低版本会出现resnet50等测试项不能通过，更新到最新版本可以通过。

可以在eigen.cmake里加一下注释。

Comment (yihuaxu) [reply cmake/external/eigen.cmake]: Done!
https://github.com/PaddlePaddle/Paddle/pull/14210/commits/5fadc8766bc61fe0fcac7f9c98e6cae4e279e4c2

**Comment by tensor-tang** [comment]
@yihuaxu  请查看comment https://github.com/PaddlePaddle/Paddle/pull/14041#issuecomment-435267368

**Comment by yihuaxu** [comment]
> @yihuaxu 请查看comment [#14041 (comment)](https://github.com/PaddlePaddle/Paddle/pull/14041#issuecomment-435267368)

用最新的代码，清掉环境能编译通过的。

**Comment by yihuaxu** [comment]
刚看了下CI 的log提示GPU内存不足，麻烦帮重新再跑一次
链接： http://ci.paddlepaddle.org/viewLog.html?buildId=26065&buildTypeId=Paddle_PrCi&tab=buildLog&_focus=14532
截图： 
image

**Comment by tensor-tang** [comment]
已经rerun

**Comment by tensor-tang** [comment]
现在测试avx512需要 revert #14239

**Comment by yihuaxu** [comment]
> 现在测试avx512需要 revert #14239

@tensor-tang 现在为避免冲突，我已经把我的AVX512检查部分Revert掉了，链接如下
https://github.com/PaddlePaddle/Paddle/pull/14210/commits/dba75a2a88048fe23fd86b9b470a1247b7c09553

**Comment by tensor-tang** [comment]
revert的主要目的是simd 的flag现在没有打开avx512了，所以如果不revert是测不到avx512的。

已经rerun了，但是最好先merge下develop。当前commit下CI没有测到AVX512.

**Comment by yihuaxu** [comment]
> revert的主要目的是simd 的flag现在没有打开avx512了，所以如果不revert是测不到avx512的。
> 
> 已经rerun了，但是最好先merge下develop。当前commit下CI没有测到AVX512.

OK

**Comment by yihuaxu** [comment]
@tensor-tang Please help us review this PR. About the last modification for test_test_parallel_executor_seresnext , I found that it be about the accuracy issue and change the range from 1e-3 to 1e-2. The below is the related log.

[04:28:02] :	 [Step 1/1] ======================================================================
[04:28:02] :	 [Step 1/1] FAIL: test_seresnext_with_learning_rate_decay (test_parallel_executor_seresnext.TestResnet)
[04:28:02] :	 [Step 1/1] ----------------------------------------------------------------------
[04:28:02] :	 [Step 1/1] Traceback (most recent call last):
[04:28:02] :	 [Step 1/1]   File "/paddle/build/python/paddle/fluid/tests/unittests/test_parallel_executor_seresnext.py", line 318, in test_seresnext_with_learning_rate_decay
[04:28:02] :	 [Step 1/1]     model=SE_ResNeXt50Small, use_cuda=False, iter=2, delta2=1e-3)
[04:28:02] :	 [Step 1/1]   File "/paddle/build/python/paddle/fluid/tests/unittests/test_parallel_executor_seresnext.py", line 313, in _check_resnet_convergence
[04:28:02] :	 [Step 1/1]     np.mean(parallel_last_loss), single_last_loss[0], delta=delta2)
[04:28:02] :	 [Step 1/1] AssertionError: 10.173218 != 10.17434 within 0.001 delta

File: cmake/configure.cmake [hunk 2]
```diff
@@ -50,7 +50,11 @@ if(NOT WITH_PROFILER)
 endif(NOT WITH_PROFILER)
 
 if(NOT CMAKE_CROSSCOMPILING)
-    if(WITH_AVX AND AVX_FOUND)
+    if(WITH_AVX AND AVX512F_FOUND)
```
Comment (tensor-tang) [inline cmake/configure.cmake:53]: We can not enable AVX512 yet, since it would cause the illegal instruction issue.

Comment (yihuaxu) [reply cmake/configure.cmake:53]: OK, got it! It can be as the reference when you solve the illegal instruction issue.

**Pull request closed without merging**